azure-pipelines-template.yml

```diff
@@ -0,0 +1,25 @@
+steps:
+- task: NodeTool@0
+  inputs:
+    versionSpec: '8.9'
+  displayName: 'Install Node.js'
+
+- script: npm install
+  displayName: 'npm install'
+
+- script: npm install -g mocha
+  displayName: 'Install mocha'
+
+- script: npm install -g mocha-junit-reporter
+  displayName: 'Install mocha junit reporter'
```

Comment:
Why the whole custom mocha install/deep test file run instead of a simple `npm test`?

Reply:
Ah, because of the xml result collection.

Reply:
I'd consider editing https://github.com/Microsoft/TypeScript-TmLanguage/blob/master/package.json#L12 instead of doing lots of tool-specific work - it's not like the xml file is going to fail any other CI providers. On that note, using _just_ the junit reporter is kind eck - it ruins the logfile output for humans (and removes all progress indication). Setting up a [multi-reporter](https://www.npmjs.com/package/mocha-multi-reporters) structure would be much better.

Reply:
Good call - I've updated appropriately with a multi-reporter structure within the npm test call